docs/docusaurus/___test___/wasThisHelpful.test.js

```diff
@@ -0,0 +1,82 @@
+import * as React from "react";
+import { test } from "@jest/globals";
+
+import { render, screen } from "@testing-library/react";
+import userEvent from '@testing-library/user-event'
+import '@testing-library/jest-dom';
+import WasThisHelpful from "../src/components/WasThisHelpful";
+
+describe('"Was this Helpful?" section', () => {
+
+    const posthog = {
+        capture: (eventName,content) => {}
+    }
+    Object.defineProperty(global.window, "posthog", {
+        value: posthog
+    })
+
+    test("Buttons should be enabled when they have not been clicked", () => {
+        render(
+            <WasThisHelpful/>
+        );
+        expect(screen.getByText('Yes')).not.toBeDisabled();
+        expect(screen.getByText('No')).not.toBeDisabled();
+    });
+
+    test("Buttons should be disabled when 'Yes' button has been clicked", async () => {
+        render(
+            <WasThisHelpful/>
+        );
+        await userEvent.click(screen.getByText('Yes'));
+
+        expect(screen.getByText('Yes')).toBeDisabled();
+        expect(screen.getByText('No')).toBeDisabled();
+    });
+
+    test("Buttons should be disabled when 'No' button has been clicked", async () => {
+        await selectingNoInWasThisHelpful();
+
+        expect(screen.getByText('Yes')).toBeDisabled();
+        expect(screen.getByText('No')).toBeDisabled();
+    });
+
+    test("Feedback Modal should pop-up when 'No' button has been clicked", async () => {
+        await selectingNoInWasThisHelpful();
+
+        expect(screen.getByTestId('feedback-modal')).toBeInTheDocument();
+    });
+
+    test("Submit button in Feedback Modal is disabled when description is blank", async () => {
+        await selectingNoInWasThisHelpful();
+
+        expect(screen.getByText("Submit")).toBeDisabled();
+    });
+
+    test("Submit button in Feedback Modal is enabled when description input is filled", async () => {
+        await selectingNoInWasThisHelpful();
+
+        const descriptionBox = screen.getByTestId('description-textbox');
```

Comment:
Yes, I agree. I tried to find the text box by name but it said that name value was empty. So I couldn't access the textbox by using screen.getByRole('textbox', { name: 'description'}), do you have any idea how I could access it?

Reply:
try ` getByPlaceholderText`

Reply:
I was not sure about placeholder text because it was a long text, but it's true that compared to having getByTestId that needs test-related code in the react component, the placeholder is a much better solution :place_of_worship: